test/await-after.test.js

```diff
@@ -247,7 +247,7 @@ test('await after complex scenario', async (t) => {
   t.notOk(fourthLoaded, 'fourth is not loaded')
   app.use(second)
   t.ok(firstLoaded, 'first is loaded')
-  t.notOk(secondLoaded, 'second is not loaded')
+  t.ok(secondLoaded, 'second is loaded')
```

Comment:
I'm not exactly sure why this behavior changed.  My guess is that delaying the plugin's promise resolution to `process.nextTick` results in a different ordering of items in the root plugin's work queue, allowing the second plugin's function to be called on the same tick as the `app.use(second)` on L250 without having to `await` to allow that function to be called on a future tick.

I defer to the maintainers if this is a concerning change or not.

Reply:
Unfortunately it is :(.

Reply:
Is the expectation here that a plugin function is _never_ called on the same tick as when it's registered w/ avvio?  i.e., similar to how the promise spec dictates that a `.then(cb)` callback is to be invoked on the next turn of the event loop, not the current